astronomer/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -151,8 +151,17 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         self.preprocess_config()
         model_config = self.config.get("Model")
         transform_config = self.config.get("Transform", self.config)
-        if self.check_if_job_exists:
-            self._check_if_transform_job_exists()
+        if self.check_if_job_exists:  # pragma: no cover
+            try:
+                # for apache-airflow-providers-amazon<=7.2.1
+                self._check_if_transform_job_exists()  # type: ignore[attr-defined]
+            except AttributeError:
+                # for apache-airflow-providers-amazon>=7.3.0
+                transform_config["TransformJobName"] = self._get_unique_job_name(
+                    transform_config["TransformJobName"],
+                    self.action_if_job_exists == "fail",
+                    self.hook.describe_transform_job,
+                )
```

Comment:
In my opinion, we should check code for the new provider version first and handle exceptions for older versions. WDYT?